learning_observer/learning_observer/auth/social_sso.py

```diff
@@ -143,6 +159,28 @@ async def social_handler(request):
     return aiohttp.web.HTTPFound(url)
 
 
+async def determine_roster_source(request):
+    """
+    Retrieve the data source type for roster data from the PMSS settings
+    """
+    roster_source = settings.pmss_settings.source(types=['roster_data'])
+    return roster_source
+
+async def _set_lms_header_information(request, roster_source):
```

Comment:
Ditto here. This is unused complexity. This kind of dispatch makes  sense to add once we have at least two LMSes. We should not call this, but rather, have this in-line above.

Reply:
Also, when we do want generic dispatch, this should handle the case where `roster_source` is *not* in the LMS map explicitly. Right now, this will fail if we call, for example, with `roster_source` set to `Canvas` or with another typo.

Reply:
I left this function here because we already have Schoology code that uses the mapping. And in the case of unknown or typo roster_source, it would have failed from the top when the settings is being loaded.